Compiler warnings when PICO_DEBUG_MALLOC enabled

Compiler is unhappy about implicit `printf` declaration if malloc debugging is enabled:
```
/usr/share/pico-sdk/src/rp2_common/pico_malloc/pico_malloc.c: In function '__wrap_malloc':
/usr/share/pico-sdk/src/rp2_common/pico_malloc/pico_malloc.c:45:9: warning: implicit declaration of function 'printf' [-Wimplicit-function-declaration]
   45 |         printf("malloc %d %p->%p\n", (uint) size, rc, ((uint8_t *) rc) + size);
      |         ^~~~~~
/usr/share/pico-sdk/src/rp2_common/pico_malloc/pico_malloc.c:13:1: note: include '<stdio.h>' or provide a declaration of 'printf'
   12 | #include "pico/mutex.h"
  +++ |+#include <stdio.h>
   13 | auto_init_mutex(malloc_mutex);
/usr/share/pico-sdk/src/rp2_common/pico_malloc/pico_malloc.c:45:9: warning: incompatible implicit declaration of built-in function 'printf' [-Wbuiltin-declaration-mismatch]
   45 |         printf("malloc %d %p->%p\n", (uint) size, rc, ((uint8_t *) rc) + size);
      |         ^~~~~~
/usr/share/pico-sdk/src/rp2_common/pico_malloc/pico_malloc.c:45:9: note: include '<stdio.h>' or provide a declaration of 'printf'
/usr/share/pico-sdk/src/rp2_common/pico_malloc/pico_malloc.c: In function '__wrap_calloc':
/usr/share/pico-sdk/src/rp2_common/pico_malloc/pico_malloc.c:62:9: warning: incompatible implicit declaration of built-in function 'printf' [-Wbuiltin-declaration-mismatch]
   62 |         printf("calloc %d %p->%p\n", (uint) (count * size), rc, ((uint8_t *) rc) + size);
      |         ^~~~~~
/usr/share/pico-sdk/src/rp2_common/pico_malloc/pico_malloc.c:62:9: note: include '<stdio.h>' or provide a declaration of 'printf'
/usr/share/pico-sdk/src/rp2_common/pico_malloc/pico_malloc.c: In function '__wrap_realloc':
/usr/share/pico-sdk/src/rp2_common/pico_malloc/pico_malloc.c:79:9: warning: incompatible implicit declaration of built-in function 'printf' [-Wbuiltin-declaration-mismatch]
   79 |         printf("realloc %p %d->%p\n", mem, (uint) size, rc);
      |         ^~~~~~
```

--- a/src/rp2_common/pico_malloc/pico_malloc.c
+++ b/src/rp2_common/pico_malloc/pico_malloc.c
@@ -13,6 +13,10 @@
 auto_init_mutex(malloc_mutex);
 #endif
 
+#if PICO_DEBUG_MALLOC
+#include <stdio.h>
+#endif
+
 extern void *REAL_FUNC(malloc)(size_t size);
 extern void *REAL_FUNC(calloc)(size_t count, size_t size);
 extern void *REAL_FUNC(realloc)(void *mem, size_t size);

